third_party/nvfuser/csrc/fusion_segmenter.cpp

```diff
@@ -1844,8 +1850,7 @@ bool TranslateApplicableWelford::isValidPersistentFusion(
 
   auto scheduler = SchedulerEntry::makeEntry(
       ScheduleHeuristic::Persistent, translated_fusion, runtime_info);
-
-  return scheduler->reductionParams().persistent_kernel;
+  return scheduler->reductionParams().persistent_kernel && scheduler->reductionParams().fastest_dim;
```

Comment:
The current logic is:  
```
  if (segmented) {
    TranslateApplicableWelford(SegmentedFusion* segmented_fusion, const KernelArgumentHolder& runtime_inputs)
  } else {
    TranslateApplicableWelford(Fusion* fusion, const KernelArgumentHolder& runtime_inputs)
  }

```
The checking code was deleted in your post is to allow persisent scheduler working with Welford ops. It doesn't include translation of Welford to two-pass approach.

Reply:
> canSchedule

canSchedule only checks if it is a persistent kernel, but here we also want to check if it is a reduction on the fatest dim. If we really want to avoid the call to `makeEntry`, then we should add the following code to check if the reduciton is on the fastest dim.
```
  //auto scheduler = SchedulerEntry::makeEntry(
  //    ScheduleHeuristic::Persistent, translated_fusion, runtime_info);
  //return scheduler->reductionParams().persistent_kernel && scheduler->reductionParams().fastest_dim;  
  auto reduction_tv_entry =
        HeuristicSummaryEntry<HeuristicCompileTime::ReductionTVs>(
            nullptr, [&translated_fusion]() {
              return std::make_unique<std::vector<TensorView*>>(
                  scheduler_utils::getReductionTvs(translated_fusion));
            });

  auto& reduction_tvs = reduction_tv_entry.get();
  auto properties =
        scheduler_utils::getProperties(translated_fusion, runtime_info, reduction_tvs[0]);
  return properties.fastest_dim_reduction;
```

Reply:
The previous approach would run the segmenter just to translate WelfordOps, which seems rather inefficient. I like your approach of run the translator in the segmenter even in the unsegmented case.